## Understanding the HMS Dataset

The purpose of this notebook is to explore the raw values obtained from the Healthy Minds Study (HMS) datasets from the years 2022 - 2026.

## Goals

- Understand the data
- Understand the study
- Identify possible factors to include
- Look for limiations
- Specify questions

## HMS

The Healthy Minds Study is survey on post-secondary student mental health, It is a web-based survey for students who are currently enrolled in post-secondary education and over the age of 18, which includes both undergraduate and graduate students.

## Dataset Structure

In [3]:
# Imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## 2022 - 2023 Data

In [4]:
# Load data
hms_23 = pd.read_csv("../data/raw/2022-2023/HMS_2022-2023_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_12688\3574909435.py:2: DtypeWarning: Columns (11: gender_text, 37: black_other_text, 43: asian_other_text, 50: his_other_text, 56: st_tps, 63: st_otherdoc_text, 78: degree_other_text, 101: covid_clsformat_text, 105: enroll_text, 135: residenc_text, 186: sib_other_text, 203: alc_binge_othgen, 240: dx_bip_4_text, 248: dx_ax_6_text, 251: dx_ocd_3, 255: dx_ocd_6_text, 260: dx_trauma_4_text, 266: dx_neurodev_4_text, 270: dx_ea_4, 273: dx_ea_6_text, 276: dx_psy_3, 277: dx_psy_4, 278: dx_psy_5, 279: dx_psy_6, 282: dx_perso_1, 283: dx_perso_2, 287: dx_perso_6, 290: dx_perso_9, 291: dx_perso_10, 292: dx_perso_11, 293: dx_perso_12, 298: dx_sa_3, 299: dx_sa_2_text, 310: disab_1_9_text, 322: talk1_8_text, 412: stim_reason_text, 418: antidep_reason_text, 422: antipsy_reason_4, 423: antipsy_reason_5, 430: antianx_reason_text, 432: mood_reason_2, 434: mood_reason_4, 436: mood_reason_text, 440: slpmed_reason_4, 442: slpmed_reason_text, 448: othmed_reason_tex

In [5]:
# Inspect
hms_23.head()

,responseid,age,sex_birth,gender_male,gender_female,gender_transm,gender_transf,gender_queer,gender_nonbin,gender_prefnoresp,...,schoolnum,inst_survey,inst_hmsyear,inst_size,inst_type,inst_public,inst_geo,inst_specart,inst_speccc,inst_gradrate
0,R_305XFCxIAD4Adso,21.0,1.0,NaN,1.0,NaN,NaN,NaN,NaN,NaN,...,1,HMS Fall 2022,2022-2023,2,4,0,1,0,0,0.44
1,R_3PzBYXlOn9d51bs,21.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,...,1,HMS Fall 2022,2022-2023,2,4,0,1,0,0,0.44
2,R_qWlLL6lkNzVPttL,41.0,1.0,NaN,1.0,NaN,NaN,NaN,NaN,NaN,...,1,HMS Fall 2022,2022-2023,2,4,0,1,0,0,0.44
3,R_1jlU5T3wBavUQmp,24.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,...,1,HMS Fall 2022,2022-2023,2,4,0,1,0,0,0.44
4,R_DwRSIFjs6RRlKpP,27.0,1.0,NaN,1.0,NaN,NaN,NaN,NaN,NaN,...,1,HMS Fall 2022,2022-2023,2,4,0,1,0,0,0.44


In [6]:
# Shape
hms_23.shape

(76406, 1620)

In [7]:
# NA values
hms_23.isna().sum()

responseid           0
age                  0
sex_birth          144
gender_male      56568
gender_female    23584
                 ...  
inst_public          0
inst_geo             0
inst_specart         0
inst_speccc          0
inst_gradrate        0
Length: 1620, dtype: int64

In [8]:
# Datatypes
hms_23.dtypes

responseid           str
age              float64
sex_birth        float64
gender_male      float64
gender_female    float64
                  ...   
inst_public        int64
inst_geo           int64
inst_specart       int64
inst_speccc        int64
inst_gradrate    float64
Length: 1620, dtype: object

The HMS data from 2022 - 2023 contains 76406 observations and 1620 attributes

## 2023 - 2024 Data

In [9]:
# Load the data
hms_24 = pd.read_csv("../data/raw/2023-2024/HMS_2023-2024_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_12688\198937019.py:2: DtypeWarning: Columns (12: gender_text, 23: sexual_text, 38: black_other_text, 44: asian_other_text, 51: his_other_text, 64: st_otherdoc_text, 109: covid_clsformat_text, 113: enroll_text, 161: field_other_tech_text, 170: living_sit_text, 234: sib_other_text, 300: internet_3_8_TEXT, 348: dx_bip_4_text, 356: dx_ax_6_text, 364: dx_ocd_6_text, 370: dx_trauma_4_text, 376: dx_neurodev_4_text, 384: dx_ea_6_text, 392: dx_psy_6_text, 405: dx_perso_11_text, 410: dx_sa_2_text, 523: stim_reason_TEXT, 529: antianx_reason_TEXT, 536: meds_w_3_text, 540: stim_prescriber_7_TEXT, 571: why_tx_8_text, 608: inf_7_text, 641: ins_ina_8_text, 688: infoseek_oth_text, 699: witness_6_text, 712: inter_y_6_text, 721: inter_n_6_text, 731: inter_n_why_9_text, 790: drug_other_text, 872: ab_emo_othcont_text, 940: hostcli_other_text, 970: excl_pers_other_text, 999: campus_oth_text, 1064: sa_occur_loc6_text, 1081: sa_loc_oth_text, 1094: sa_who_oth_text, 1

In [10]:
# Inspect
hms_24.head()

,StartDate,RecordedDate,responseid,age,sex_birth,gender_male,gender_female,gender_queer,gender_nonbin,gender_trans,...,inst_size,inst_type,inst_public,inst_geo,inst_specart,inst_speccc,inst_gradrate,inst_msi,peercohort,Random
0,16nov2023 22:51:41,05dec2023 08:17:37,R_006AKYXyGhT9Upb,18.0,1.0,NaN,1.0,NaN,NaN,NaN,...,4,3,1,8,0,0,0.25,0,3.0,0.719747
1,16nov2023 17:12:11,16nov2023 17:37:33,R_006rfKhxwocpzmF,18.0,1.0,NaN,1.0,NaN,NaN,NaN,...,4,3,1,8,0,0,0.25,0,3.0,0.219248
2,24sep2023 18:29:45,24sep2023 18:53:25,R_009nvDs0qZwqoW5,18.0,2.0,1.0,NaN,NaN,NaN,NaN,...,5,3,1,5,0,0,0.50,0,3.0,0.756233
3,30oct2023 08:01:53,21nov2023 07:17:23,R_00R9rAHb09Ydma5,20.0,1.0,NaN,1.0,NaN,NaN,NaN,...,4,3,1,3,0,0,0.24,0,3.0,0.133426
4,24oct2023 09:49:54,21nov2023 07:57:58,R_00SNQFMZ5Kok7Hr,21.0,2.0,1.0,NaN,NaN,NaN,NaN,...,2,4,0,5,0,0,0.57,0,1.0,0.773790


In [11]:
# Shape
hms_24.shape

(104729, 1554)

In [12]:
# NA values
hms_24.isna().sum()

StartDate          0
RecordedDate       0
responseid         0
age                0
sex_birth        139
                ... 
inst_speccc        0
inst_gradrate      0
inst_msi           0
peercohort       107
Random           621
Length: 1554, dtype: int64

In [13]:
# Datatypes
hms_24.dtypes

StartDate            str
RecordedDate         str
responseid           str
age              float64
sex_birth        float64
                  ...   
inst_speccc        int64
inst_gradrate    float64
inst_msi           int64
peercohort       float64
Random           float64
Length: 1554, dtype: object

The HMS data from 2023 - 2024 contains 104729 observations and 1554 variables

## 2024 - 2025 Data

In [14]:
# Load the data
hms_25 = pd.read_csv("../data/raw/2024-2025/HMS_2024-2025_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_12688\1860412060.py:2: DtypeWarning: Columns (13: gender_text, 25: sexual_text, 53: state_9thgrade, 54: state_10thgrade, 55: state_11thgrade, 56: state_12thgrade, 57: year_9thgrade, 58: year_10thgrade, 59: year_11thgrade, 60: year_12thgrade, 881: Grad, 1475: Peer, 1590: StartDate2) have mixed types. Specify dtype option on import or set low_memory=False.
  hms_25 = pd.read_csv("../data/raw/2024-2025/HMS_2024-2025_PUBLIC_instchars.csv")


In [15]:
# Inspect
hms_25.head()

,StartDate,Finished,RecordedDate,responseid,schoolnum,age,gender_male,gender_female,gender_queer,gender_nonbin,...,inst_survey,inst_hmsyear,inst_size,inst_type,inst_public,inst_geo,inst_specart,inst_speccc,inst_gradrate,weighted_sum
0,29oct2024 10:39:50,1,29oct2024 11:08:57,R_1TGFyxLOqW1PZnL,1,18,1.0,NaN,NaN,NaN,...,HMS Fall 2024,2024-2025,4,3,1,3,0,0,0.16,1020.2438
1,11nov2024 09:22:49,0,19nov2024 07:28:14,R_3WBLeqnaO3RsWxr,1,19,NaN,1.0,NaN,NaN,...,HMS Fall 2024,2024-2025,4,3,1,3,0,0,0.16,1020.2438
2,15nov2024 12:45:25,0,19nov2024 07:28:18,R_3lz7HbBX7qZq3mG,1,18,NaN,NaN,NaN,NaN,...,HMS Fall 2024,2024-2025,4,3,1,3,0,0,0.16,1020.2438
3,04nov2024 09:38:47,1,04nov2024 10:04:25,R_6cwKNvO8lKIh8NX,1,18,NaN,1.0,NaN,NaN,...,HMS Fall 2024,2024-2025,4,3,1,3,0,0,0.16,1020.2438
4,18nov2024 10:18:38,0,19nov2024 07:28:18,R_5lVSZKBrDqkU8UV,1,23,NaN,NaN,NaN,NaN,...,HMS Fall 2024,2024-2025,4,3,1,3,0,0,0.16,1020.2438


In [16]:
# Shape
hms_25.shape

(84735, 1608)

In [17]:
# NA values
hms_25.isna().sum()

StartDate        0
Finished         0
RecordedDate     0
responseid       0
schoolnum        0
                ..
inst_geo         0
inst_specart     0
inst_speccc      0
inst_gradrate    0
weighted_sum     0
Length: 1608, dtype: int64

In [18]:
# Datatypes
hms_25.dtypes

StartDate            str
Finished           int64
RecordedDate         str
responseid           str
schoolnum          int64
                  ...   
inst_geo           int64
inst_specart       int64
inst_speccc        int64
inst_gradrate    float64
weighted_sum     float64
Length: 1608, dtype: object

The HMS data from 2024 - 2025 contains 84735 observations and 1608 variables

## 2025 - 2026 Data

In [19]:
# Load data
hms_26 = pd.read_csv("../data/raw/2025-2026/HMS_2025-2026_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_12688\1234493437.py:2: DtypeWarning: Columns (12: gender_text, 24: sexual_text, 55: race_other_text, 61: black_other_text, 67: asian_other_text, 74: his_other_text, 81: pi_other_text, 85: residenc_text, 87: living_sit_text, 115: clsformat_text, 131: degree_other_tech, 135: enroll_text, 157: field_other_text, 184: field_other_tech_text, 255: sib_other_text, 318: recovery_path_text, 328: recovery_service_text, 337: recovery_aware_text, 359: dx_dep_4_text, 365: dx_bip_4_text, 373: dx_ax_6_text, 381: dx_ocd_6_text, 387: dx_trauma_4_text, 393: dx_neurodev_4_text, 401: dx_ea_6_text, 409: dx_psy_6_text, 422: dx_perso_11_text, 427: dx_sa_2_text, 444: ADHD2_4_TEXT, 446: ADHD3_text, 463: A988_followup_text, 465: A988_reason_text, 478: talk1_8_text, 502: ADHD_therapy_loc_text, 575: stim_reason_TEXT, 581: antianx_reason_TEXT, 588: meds_w_3_text, 601: ADHD_meds_text, 622: why_tx_6_text, 623: why_tx_8_text, 624: why_tx_12_text, 663: inf_7_text, 696: ins_in

In [20]:
# Inspect
hms_26.head()

,StartDate,Finished,RecordedDate,responseid,age,gender_male,gender_female,gender_nonconf,gender_nonbin,gender_trans,...,inst_hmsyear,inst_size,inst_type,inst_public,inst_geo,inst_specart,inst_speccc,inst_gradrate,inst_msi,inst_msi_type
0,23oct2025 14:53:45,1,23oct2025 15:11:53,R_5iqeaawZEhbxeOR,21,1.0,NaN,NaN,NaN,1.0,...,2025-2026,1,5,0,3,1,0,0.41,0,NaN
1,18oct2025 07:55:43,0,11nov2025 11:46:28,R_30Sasd5LgdKi6P9,18,NaN,1.0,NaN,NaN,NaN,...,2025-2026,1,5,0,3,1,0,0.41,0,NaN
2,13oct2025 16:19:54,1,13oct2025 16:48:33,R_3hgejZYliaEhZEl,21,NaN,NaN,NaN,1.0,NaN,...,2025-2026,1,5,0,3,1,0,0.41,0,NaN
3,13oct2025 14:42:36,1,13oct2025 22:45:11,R_7omJcX1bOlDPm8h,21,NaN,1.0,NaN,NaN,NaN,...,2025-2026,1,5,0,3,1,0,0.41,0,NaN
4,29oct2025 20:07:21,1,29oct2025 20:45:52,R_1V4oEjNN613yO7s,20,NaN,NaN,NaN,NaN,NaN,...,2025-2026,1,5,0,3,1,0,0.41,0,NaN


In [21]:
# Shape
hms_26.shape

(67477, 1810)

In [22]:
# NA values
hms_26.isna().sum()

StartDate            0
Finished             0
RecordedDate         0
responseid           0
age                  0
                 ...  
inst_specart         0
inst_speccc          0
inst_gradrate        0
inst_msi             0
inst_msi_type    47230
Length: 1810, dtype: int64

In [23]:
# Datatypes
hms_26.dtypes

StartDate            str
Finished           int64
RecordedDate         str
responseid           str
age                int64
                  ...   
inst_specart       int64
inst_speccc        int64
inst_gradrate    float64
inst_msi           int64
inst_msi_type        str
Length: 1810, dtype: object

The HMS data from 2025 - 2026 contains 67477 observations and 1810 variables

## Comparisons Between Data

In [24]:
# Difference between rows and columns
hms_summary = pd.DataFrame(
    {"Year": ["2022-2023", "2023-2024", "2024-2025", "2025-2026"],
    "Rows": [hms_23.shape[0], hms_24.shape[0], hms_25.shape[0], hms_26.shape[0]],
    "Columns": [hms_23.shape[1], hms_24.shape[1], hms_25.shape[1], hms_26.shape[1]]}
)
hms_summary

,Year,Rows,Columns
0,2022-2023,76406,1620
1,2023-2024,104729,1554
2,2024-2025,84735,1608
3,2025-2026,67477,1810


The amount of observations and variables varies between the years with 2023-2024 having the most participants and 2025-2026 having the least amount of participants. The variation of columns means that the structure of the survey has changed from year-to-year. Before combining, cleaning, and analyzing the datasets further, the variables will need to be compared across each year.

In [25]:
# Compare columns across years

# Columns
hms_23_cols = hms_23.columns.to_list()
hms_24_cols = hms_24.columns.to_list()
hms_25_cols = hms_25.columns.to_list()
hms_26_cols = hms_26.columns.to_list()

# Compare using set intersection
common_cols = list(set(hms_23_cols) & set(hms_24_cols) & set(hms_25_cols) & set(hms_26_cols))
len(common_cols)

1067

The datasets share 1067 variables across all four years the survey was done. These variables will act as a starting point for identifying factors and outcome that will be used in the combined analysis of the four years. Although these variables share the same name, they must have the same meaning, essentially the same survey question, and same response coding across all four years. This analysis will be done using the data dictionaries from the HMS codebooks from 2022-2023, 2023-2024, 2024-2025, and 2025-2026. Variables will be split into: Outcome, Main factor, control, identifier, and survery weight. Outcome variables are ones we are trying to explain, main factors are going to be used to see if it is associated with the outcome, control variables are background charasteristics that can be accounetd for when analyzing relationships between main factors and outcomes, identifiers will be used to know where the observation came from, and survey weights will be kept for use later in statistical analysis to account for survey design and produce weighted estimates.

## Data Dictionary

| Variable | Category | Role | Description | Coding / Range |
|---|---|---|---|---|
| `flourish` | Mental well-being | Outcome | Flourishing Scale total score measuring positive mental well-being | 8–56; higher values indicate greater flourishing |
| `deprawsc` | Depression | Outcome | PHQ-9 total depression symptom score | 0–27; higher values indicate greater depressive symptom severity |
| `anx_score` | Anxiety | Outcome | GAD-7 total anxiety symptom score | 0–21; higher values indicate greater anxiety symptom severity |
| `fincur` | Financial | Main factor | Self-reported current financial stress | 1 = Always stressful, 2 = Often stressful, 3 = Sometimes stressful, 4 = Rarely stressful, 5 = Never stressful |
| `finpast` | Financial | Main factor | Self-reported financial stress while growing up | 1 = Always stressful, 2 = Often stressful, 3 = Sometimes stressful, 4 = Rarely stressful, 5 = Never stressful |
| `belong1` | Social / Campus | Main factor | Agreement with the statement "I see myself as a part of the campus community" | 1 = Strongly agree through 6 = Strongly disagree |
| `discrim_race` | Campus experience | Main factor | Experienced unfair treatment at school because of race/ethnicity in the past 12 months | 0 = No, 1 = Yes |
| `discrim_culture` | Campus experience | Main factor | Experienced unfair treatment at school because of cultural background in the past 12 months | 0 = No, 1 = Yes |
| `discrim_gender` | Campus experience | Main factor | Experienced unfair treatment at school because of gender in the past 12 months | 0 = No, 1 = Yes |
| `discrim_sexual` | Campus experience | Main factor | Experienced unfair treatment at school because of sexual orientation in the past 12 months | 0 = No, 1 = Yes |
| `alc_any` | Behavioural | Main factor | Any alcohol consumption during the past 2 weeks | 0 = No, 1 = Yes |
| `age` | Demographic | Control | Respondent age | Age in years |
| `international` | Demographic | Control | Whether the respondent is an international student | 0 = No, 1 = Yes |
| `educ_par1` | Socioeconomic | Control | Highest education level completed by first parent, step-parent, or guardian | 1 = 8th grade or lower through 7 = Graduate degree; 8 = Don't know |
| `educ_par2` | Socioeconomic | Control | Highest education level completed by second parent, step-parent, or guardian | 1 = 8th grade or lower through 7 = Graduate degree; 8 = Don't know |
| `enroll` | Academic | Control | Respondent's enrollment status | 1 = Full-time student, 2 = Part-time student, 3 = Other |
| `survey_year` | Dataset | Identifier / Control | Academic year in which the respondent completed the HMS survey | 2022–2023, 2023–2024, 2024–2025, or 2025–2026 |
| `nrweight` | Survey design | Survey weight | HMS non-response weight used when producing weighted estimates | Continuous survey weight; not used as a predictor |.